### PR TITLE
Create the Course app

### LINKED ISSUE #40: Missing Review Entity feature and course app
Creating review database for a course app.
The review model defines the Reviews that needs to be created, so we can create and watch reviews of teachers.

### PR DESCRIPTION
**Pull request by shirazyomtov**
The following changes were made:

Create a course app.

Adding the course app to installed_apps in setting.py in the HighTeach directory.

Added the Review model to models.py

Added ReviewManager model manager to models.py

Register the Review app in admin.py.

Create migration files for initializing the data model

When User and Teacher_course PR merge, I'll make the necessary changes to the models.py.
Resolve #40

### CODE REVIEW

**Review by idan1356 — DISMISSED** [review]
LGTM

**Review by shacharlevy — APPROVED** [review]
LGTM

**Comment by Omeramsc** [comment]
Not completely related, but please add the ERD as an image file too, so it'll be visible from the repository without the need to clone it.

File: course/models.py
Comment (natifridman) [inline course/models.py:7]: What is the advantage of implementing the model manager and not using the methods in the `Review` class?

File: course/models.py
Comment (natifridman) [inline course/models.py:29]: I think this will allow multiple choices for the rating. What do you think about something like this?
```suggestion
    rating = models.IntegerField(validators=[MaxValueValidator(5),MinValueValidator(1)])
```

**Review by natifridman — COMMENTED** [review]
Looks good, I added some comments.
We usually don't merge commented code, but since most of it is auto-generated by Django I think we can merge it with the comments

File: course/models.py
Comment (shirazyomtov) [reply course/models.py:7]: The two functions I wrote contain the filter function and the query result can contain multiple instances
The query is not about a specific instance because of that it is better to use the manager.

File: course/models.py
Comment (shirazyomtov) [reply course/models.py]: > I think this will allow multiple choices for the rating. What do you think about something like this?

Yes, I think it's more correct in this way, I fixed it.

File: course/models.py
Comment (natifridman) [inline course/models.py:26]: Looks like this is not needed anymore

**Review by Yardengz — APPROVED** [review]

**Review by natifridman — CHANGES_REQUESTED** [review]

File: course/models.py
Comment (shirazyomtov) [reply course/models.py]: > Looks like this is not needed anymore

Fixed

**Review by natifridman — APPROVED** [review]
LGTM, do you want to wait for the user model before merge?

**Comment by shirazyomtov** [comment]
> LGTM, do you want to wait for the user model before merge?

I think it is possible to merge and when the user is merged I will upload a PR with the tests and the
necessary changes.

**Comment by natifridman** [comment]
> I think it is possible to merge and when the User and Teacher Course is merged I will upload a PR with the tests and the necessary changes.

Ok, could you please squash the commits?

**Comment by shirazyomtov** [comment]
> > I think it is possible to merge and when the User and Teacher Course is merged I will upload a PR with the tests and the necessary changes.
> 
> Ok, could you please squash the commits?

Done

**Review by natifridman — DISMISSED** [review]

**Comment by shirazyomtov** [comment]
> Please resolve conflicts

Done

**Review by natifridman — APPROVED** [review]

**Review by idan1356 — APPROVED** [review]
LGTM

**Review by Yardengz — APPROVED** [review]

**Pull request merged**
